### PR TITLE
Clean up code sample and other docs

### PR DESCRIPTION
**Pull request by frant-hartm**
Doc updates originally part of #2208, but unrelated to grpc how to guide.

### CODE REVIEW

File: site/docs/how-tos/custom-batch-source.md [hunk 1]
```diff
@@ -1,28 +1,25 @@
 ---
-title: Create a Batch Source
+title: Add Batching to Custom Source
```
Comment (cangencer) [inline site/docs/how-tos/custom-batch-source.md:2]: title seems wrong, it's about creating a batch source, not adding batching to a source

File: site/docs/api/sources-sinks.md [hunk 2]
```diff
@@ -1097,7 +1098,8 @@ Pipeline p = Pipeline.create();
 p.readFrom(TestSources.items("a", "b", "c", "d"))
  .writeTo(Sinks.observable(observable));
 
-Future<List<String>> future = observable.toFuture(s -> s.collect(Collectors.toList()));
+Future<List<String>> future = observable.toFuture(
+    s -> s.collect(Collectors.toList()));
```
Comment (cangencer) [inline site/docs/api/sources-sinks.md:1102]: can we align the `);` on the next line? it's easier to read like that.

**Comment by cangencer** [comment]
Noting also that you can use `docs-no-build` label for docs only PRs to skip the full PR builder

File: site/docs/how-tos/custom-batch-source.md (hunk 1 above)
Comment (mtopolnik) [reply site/docs/how-tos/custom-batch-source.md:2]: Actually the only new point this How-To introduces is the batching. Originally there was a tutorial, then it got split into the "basics" which are in _Sources and Sinks_ and this how-to, which explains how to add batching.

Comment (cangencer) [reply site/docs/how-tos/custom-batch-source.md:2]: ok, but it's confusing if user is trying to look for how to create a custom source, not just add batching to it, where 's the entry point?

Comment (mtopolnik) [reply site/docs/how-tos/custom-batch-source.md:2]: In the section titled "Sources and Sinks", subsection "Custom Sources and Sinks".

**Pull request merged**